### PR TITLE
feat: bump brain tests to v0.0.12

### PR DESCRIPTION
**Pull request by f0rmiga**
## Description

Version 0.0.12 of the brain tests include a mechanism to skip tests based on a BOSH property. Bumping this release on KubeCF also introduces the conditional skipping based on some components being enabled or not on KubeCF.

## Motivation and Context

Improving brain tests to make it pass with KubeCF.

## How Has This Been Tested?

Locally.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code has security implications.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: deploy/helm/kubecf/assets/operations/instance_groups/brain-tests.yaml [hunk 1]
```diff
@@ -37,6 +37,15 @@
           password: ((cf_admin_password))
           skip_ssl_validation: true
           user: admin
+        tests:
+          autoscaler:
+            enabled: {{ .Values.features.autoscaler.enabled }}
+          tcprouting:
+            enabled: {{ .Values.features.routing_api.enabled }}
+          insecure_registry:
```
Comment (mook-as) [inline deploy/helm/kubecf/assets/operations/instance_groups/brain-tests.yaml:45]: Technically this should be fixed in the test script instead, or have it fail fast when it hits an invalid configuration. But whatever, that's okay too.

Comment (f0rmiga) [reply deploy/helm/kubecf/assets/operations/instance_groups/brain-tests.yaml:45]: It's the responsibility of KubeCF to tell the test what should and what should not be executed.

Comment (mook-as) [reply deploy/helm/kubecf/assets/operations/instance_groups/brain-tests.yaml:45]: It's an implementation detail that the insecure registry test depends on TCP routing, though?

**Comment by mook-as** [comment]
This deployed fine on [diego](https://concourse.suse.dev/builds/10681) and [eirini](https://concourse.suse.dev/builds/10682), it's good for merge (but I can't because I need to wait for all of CI to pass before I'm allowed to click the merge button).

**Pull request merged**